openassessment/xblock/static/js/src/oa_server.js

```diff
@@ -612,6 +612,43 @@ if (typeof OpenAssessment.Server === "undefined" || !OpenAssessment.Server) {
             $.ajax({
                 type: "POST", url: url, data: payload, contentType: jsonContentType
             });
-        }
-    };
+        },
+
+        /**
+         * 
+         */
+        getTeamDetail: function(teamId) {
+            var teamsUrl = 'http://www.courses.edx.org/api/team/v0/teams/`${teamId}`/'
+            return $.ajax({
+                type: "GET",
+                url: teamsUrl,
+                contentType: jsonContentType
+            })
+        },
+
+        listTeams: function(courseId) {
+            var teamsUrl = 'http://www.courses.edx.org/api/team/v0/teams/';
+            return $.Deferred(function(defer) {
+                $.ajax({
+                    type: "GET",
+                    url: teamsUrl,
+                    data: JSON.stringify({course_id: courseId}),
+                    contentType: jsonContentType
+                }).done(function(data) {
```

Comment:
I am confused - why can't we have multiple teams in a given course?

Reply:
from the ticket: 

"Use this view to determine the team identifier for the current user and course - you will never get back a list of length > 1 with this call (this will change in the medium-term future): "

"JS unit tests - required.  Cover the basic cases: no team, one team, what if I erroneously get more than 1 team back from the API endpoint?  What if the API returns a non-200 status code response?  What if the waffle switch is not enabled."